api_v3/services/SessionService.php

```diff
@@ -128,7 +128,11 @@ function impersonateAction($secret, $impersonatedPartnerId, $userId = "", $type
 		{
 			$impersonatedSecret = $impersonatedPartner->getSecret();
 		}
-		
+
+		if ($partnerId < 0)
```

Comment:
Why is this added here?
There is already a place in the code where we manipulate the KS max expiry levels and it's here:
https://github.com/kaltura/server/blob/38a1a0ec7270ad98048eb29ca1c7927bae2a96f4/alpha/lib/model/Partner.php#L135

Reply:
from what i saw, that place only validates the secret and expiry, but doesn't use it actually when starting the session.

Reply:
It depends on what you pass when calling startKSession if you don't pass $enforcePartnerKsMaxExpiry it wont actually do anything 
Either way if your touching the ks expiry the more correct place to do it is where all this logic is and not in the service call itself

Reply:
i see, ok. i'll look at it